pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -342,9 +348,36 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 		}
 	}
 	if cpuset, ok := s.GetCPUSet(string(pod.UID), container.Name); ok {
-		p.updateCPUsToReuse(pod, container, cpuset)
-		klog.InfoS("Static policy: container already present in state, skipping", "pod", klog.KObj(pod), "containerName", container.Name)
-		return nil
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			klog.InfoS("Static policy: container already present in state, attempting InPlacePodVerticalScaling", "pod", klog.KObj(pod), "containerName", container.Name)
+			cpusInUse := getAssignedCPUsOfSiblings(s, string(pod.UID), container.Name)
+			s.Delete(string(pod.UID), container.Name)
+			toRelease := cpuset.Difference(cpusInUse)
+			previousdefaultcpuset := s.GetDefaultCPUSet()
+			s.SetDefaultCPUSet(s.GetDefaultCPUSet().Union(toRelease))
+			p.updateCPUsToReuse(pod, container, cpuset)
```

Comment:
why do we need (or prefer to) release the full set instead of trying to add the delta?

Reply:
I  am re-evaluating the approach, based on your previous comments. Will come with a newer push, will add back the WIP label. Thanks your comments helped.

Reply:
I for myself I like more the all-in-one approach, but I'm not a fan of managing the global state this way, with undo/redo logic. I wonder if we can compute a new possible state and replace it only if succesful, avoiding the undo/redo cycle.
Something like
```
candidate := m.state.Clone()
// remove current allocation from candidate state
// attempt new allocation against candidate state
// if allocation succesfull, replace the current state with the candidate state
// store the updated state to the checkpoint file
```
In any case, we need to make sure to have a good way to observe the changes here, otherwise troubleshooting will be a nightmare. Logs at minimum can use some improvements (in general).

Reply:
Hi @ffromani , working on your proposal, i will try to improve this.

Reply:
Hi i did some tests, I managed to achieve using the delta enabling `distribute-cpus-across-cores ` static policy option. Current implementation i think follows your proposal since i use the policy_static state in memory, topology manager response is the moment of truth. In this fix i have intentionally avoided modifying topoloigy_manager code to put extra control ( i.e. i am only using the undo/redo logic in the state in memory which is also used already today when a pod is deleted ). Static policy options combinations are getting bigger, i gave it a try to improve the test coverage in this fix, but i think this will need to be discussed also as part of the CPUManager e2e_node tests improvements which has been raised some time ago.
![img_0812](https://github.com/user-attachments/assets/4fcf3555-540c-45bd-bcec-4230af8715a2)

Reply:
hi again, i discovered that current implementation behaves differently ( with regard delta ) with the ACPI MADT table ( MadtCoreEnumeration) Bios settings which impact the end CPU numbering in short it seems with linear it uses the delta with linear but with round robin it has some issues. I will dig more into this.

Reply:
[Update]  reuse of the delta requires modification of cpu_assignment.go , i am working on a prototype, i will update commit soon !